client/driver/lxc.go

```diff
@@ -0,0 +1,486 @@
+//+build linux
+
+package driver
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/nomad/client/config"
+	"github.com/hashicorp/nomad/client/fingerprint"
+	"github.com/hashicorp/nomad/client/stats"
+	"github.com/hashicorp/nomad/helper/fields"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/mitchellh/mapstructure"
+
+	dstructs "github.com/hashicorp/nomad/client/driver/structs"
+	cstructs "github.com/hashicorp/nomad/client/structs"
+	lxc "gopkg.in/lxc/go-lxc.v2"
+)
+
+const (
+
+	// containerMonitorIntv is the interval at which the driver checks if the
+	// container is still alive
+	containerMonitorIntv = 2 * time.Second
+)
+
+var (
+	LXCMeasuredCpuStats = []string{"System Mode", "User Mode", "Percent"}
+	LXCMeasuredMemStats = []string{"RSS", "Cache", "Swap", "Max Usage", "Kernel Usage", "Kernel Max Usage"}
+)
+
+// Add the lxc driver to the list of builtin drivers
+func init() {
```

Comment:
@dadgar So that we don't have to build the driver on other platforms. This driver needs the lxc-dev package which is unavailable on other platforms.